shell/platform/embedder/embedder.h

```diff
@@ -831,6 +831,85 @@ typedef struct {
   };
 } FlutterRendererConfig;
 
+/// Display refers to a graphics hardware system consisting of a framebuffer,
+/// typically a monitor or a screen. This ID is unique per display and is
+/// stable until the Flutter application restarts.
+typedef uint64_t FlutterEngineDisplayId;
+
+typedef struct {
+  /// The size of this struct. Must be sizeof(FlutterWindowMetricsEvent).
+  size_t struct_size;
+  /// Physical width of the window.
+  size_t width;
+  /// Physical height of the window.
+  size_t height;
+  /// Scale factor for the physical screen.
+  double pixel_ratio;
+  /// Horizontal physical location of the left side of the window on the screen.
+  size_t left;
+  /// Vertical physical location of the top of the window on the screen.
+  size_t top;
+  /// Top inset of window.
+  double physical_view_inset_top;
+  /// Right inset of window.
+  double physical_view_inset_right;
+  /// Bottom inset of window.
+  double physical_view_inset_bottom;
+  /// Left inset of window.
+  double physical_view_inset_left;
+  /// The identifier of the display the view is rendering on.
+  FlutterEngineDisplayId display_id;
+  /// The view that this event is describing.
+  int64_t view_id;
+} FlutterWindowMetricsEvent;
+
+typedef struct {
+  /// The size of this struct.
+  /// Must be sizeof(FlutterAddViewResult).
+  size_t struct_size;
+
+  /// True if the add view operation succeeded.
+  bool added;
+
+  /// The |FlutterAddViewInfo.user_data|.
+  void* user_data;
+} FlutterAddViewResult;
+
+/// The callback invoked by the engine when the engine has attempted to add a
+/// view.
+///
+/// The |FlutterAddViewResult| will be deallocated once the callback returns.
```

Comment:
> will be deallocated

Consider clarifying by whom, or another wording might be that the `FlutterAddViewResult` is only guaranteed to be valid during the callback itself.

Reply:
I like your new wording, switched to that!